lib/api/funnel.js

```diff
@@ -493,76 +508,55 @@ class Funnel {
    * @param {Request} request
    * @returns {Promise}
    */
-  executePluginRequest (request) {
-    return Bluebird.resolve()
-      .then(() => doAction(this.getController(request), request))
-      .catch(e => {
-        this.handleErrorDump(e);
-        return Bluebird.reject(e);
-      });
+  async executePluginRequest (request) {
+    try {
+      return await doAction(this.getController(request), request);
```

Comment:
Why do you await the promise before returning a new one ?

Reply:
to catch exceptions thrown by doAction